src/server/game/Maps/Map.cpp

```diff
@@ -2727,9 +2727,11 @@ bool Map::ActiveObjectsNearGrid(NGridType const& ngrid) const
             return true;
     }
 
-    for (ActiveNonPlayers::const_iterator iter = m_activeNonPlayers.begin(); iter != m_activeNonPlayers.end(); ++iter)
+    // Copied method from Update() to supposedly handle object removal during loop
+    for (ActiveNonPlayers::const_iterator iter = m_activeNonPlayers.begin(); iter != m_activeNonPlayers.end();)
     {
         WorldObject* obj = *iter;
+        ++iter;
```

Comment:
in which case could this change change the behavior of the code (and fix an issue) ? To me it looks like all functions called in this for loop can't modify m_activeNonPlayers